nemo/collections/multimodal/speech_llm/data/lhotse_dataset.py

```diff
@@ -124,9 +132,96 @@ def __getitem__(self, all_cuts: CutSet) -> dict[str, torch.Tensor | list[str] |
             )
             ans.update(text_minibatch)
 
+        def post_process_seq(ans):
+            max_seq_length = self.max_seq_length - 1  # reserve for eos
+            if len(ans.input_ids) > max_seq_length:
```

Comment:
I believe this truncation is a very, very bad practice and can possibly break your model without you knowing it. Consider the following scenario:

* 50% of training data exceeds the max_seq_length threshold
* you truncate the answers in 50% of cases
* after every such case, you add an <eos> prediction
* therefore, you are training the model to output incomplete answers and stop the prediction after random tokens.

In my opinion this should only be enabled if you set an option `allow_unsafe_answer_truncate` that is always `False` by default. In EMMeTT branch I have a filter for `max_tokens` that allows you to filter out these examples instead.